src/ansys/mapdl/core/cli/hpc.py

```diff
@@ -0,0 +1,271 @@
+    default=None,
+    type=str,
+    help="""If desired, you can provide a shell script to execute instead of
+the python file. You can call your python file from it if you wish. By default,
+it is not used.""",
+)
+@click.option(
+    "--requirements_file",
+    default=None,
+    type=str,
+    help="""If provided, the created virtual environment is installed with the
+libraries specified in this file. If not, the activated virtual environment is
+cloned through a temporary 'pip list' file. If you are using editable package,
+it is recommended you attach your own requirement file using ``pip freeze`` """,
+)
+@click.option(
+    "--extra_files",
+    default=None,
+    type=str,
+    help="""To upload extra files which can be called from your main python file
+(or from the shell file).""",
+)
+@click.option(
+    "--config_file",
+    default=None,
+    type=str,
+    help="""To load job configuration from a file.""",
+)
+@click.option(
+    "--num_cores",
+    default=None,
+    type=str,
+    help="""Set the amount of CPU cores reserved for the job. By default it is 1 CPU.""",
+)
+@click.option(
+    "--memory",
+    default=None,
+    type=str,
+    help="""Set the amount of memory RAM in MB reserved for the job. By default it is 100 MB.""",
+)
+@click.option(
+    "--disk_space",
+    default=None,
+    type=str,
+    help="""Set the amount of hard drive space in MB reserved for the job. By default it is 100 MB.""",
+)
+@click.option(
+    "--exclusive",
+    default=None,
+    type=str,
+    is_flag=False,
+    flag_value=True,
+    help="""Set the job to run in a machine exclusively, without sharing it with other jobs. By default it is False""",
+)
+@click.option(
+    "--max_execution_time",
+    default=None,
+    type=str,
+    help="""Set the maximum execution time for the job. By default it is zero (unlimited).""",
+)
+@click.option(
+    "--wait",
+    default=None,
+    type=str,
+    is_flag=False,
+    flag_value=True,
+    help="""Set the terminal to wait for job completion before return the control to the user. """,
+)
+@click.option(
+    "--save_config_file",
+    default=False,
+    type=bool,
+    is_flag=False,
+    flag_value=True,
+    help="""Writes the configuration to the config file, after successfully
+submit the job. It overwrites the configuration file.
+The configuration file path is given using ``config_file`` argument.""",
+)
+@click.option(
+    "--debug",
+    default=False,
+    type=bool,
+    is_flag=False,
+    flag_value=True,
+    help="""Set PyMAPDL to prints the debug logging to the console output.""",
+)
+def submit(
+    main_file: str,
+    name: str = None,
+    url: str = None,
+    user: str = None,
+    password: str = None,
+    python: float = 3.9,
+    output_files: Optional[Union[str, list]] = None,
+    shell_file: str = None,
+    requirements_file: str = None,
+    extra_files: Optional[Union[str, list]] = None,
+    config_file: str = None,
+    num_cores: int = None,
```

Comment:
option to download the result files?

Reply:
I am not aware if we can do that from the PyHPS library... i will check it out.

Reply:
Yes, It can be done.

Reply:
https://hps.docs.pyansys.com/version/stable/api/_autosummary/ansys.hps.client.jms.api.ProjectApi.download_file.html#download-file